CoreDataStack/Entity/Contact.swift

```diff
@@ -142,3 +143,19 @@ extension Contact: Managed {
         return [NSSortDescriptor(keyPath: \Contact.name, ascending: true)]
     }
 }
+
+extension Contact {
+    
+    public static var notIdentityPredicate: NSPredicate {
+        return NSPredicate(format: "%K != nil AND %K.%K == nil", #keyPath(Contact.keypair), #keyPath(Contact.keypair), #keyPath(Keypair.privateKey))
```

Comment:
I'm not sure the underlay SQL is the same or not. The first relationship "%K != nil" could fail fast. I needs have a try later.

Reply:
`*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'keypath keypair.keypair not found in entity <NSSQLEntity Contact id=4>'`

The relationship should be check nil before use. Otherwise, the `NSInvalidArgumentException` will be throw.